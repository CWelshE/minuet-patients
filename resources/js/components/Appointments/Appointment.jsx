import React from "react";
import PropTypes from "prop-types";

/**
 * A single Appointment in an Appointments list.
 * Presentational component only.
 */
const Appointment = (props) => (
  <div className={"appointment"}>
    <h3>{props.patientName}</h3>
    <span>Date: {props.date}</span>
    <span>Time: {props.time}</span>
    <span>Type: {props.type}</span>
  </div>
);

Appointment.propTypes = {
  patientName: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default Appointment;
