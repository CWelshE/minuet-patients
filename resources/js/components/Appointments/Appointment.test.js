import React from "react";
import renderer from "react-test-renderer";
import Appointment from "./Appointment.jsx";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// Start Enzyme adapter for React
configure({ adapter: new Adapter() });

it("renders equally compared to snapshot", () => {
  const apptProps = {
    apptNum: 1,
    date: "03-24-1994",
    time: "15:30",
    type: "Vaccination",
  };
  const tree = renderer.create(<Appointment {...apptProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("has info from props", () => {
  const appt = shallow(
    <Appointment
      apptNum={2}
      date={"12/20/1990"}
      time={"1:00 PM"}
      type={"Short Visit"}
    />
  );
  expect(appt.find("h3").html()).toBe("<h3>Appointment 2</h3>");
});
