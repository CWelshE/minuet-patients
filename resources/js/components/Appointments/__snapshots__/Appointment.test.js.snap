// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders equally compared to snapshot 1`] = `
<div
  className="appointment css-1yrphkm-Appointment"
>
  <h3>
    Appointment 
    1
  </h3>
  <span>
    Date: 
    03-24-1994
  </span>
  <span>
    Time: 
    15:30
  </span>
  <span>
    Type: 
    Vaccination
  </span>
</div>
`;
